_drafts/introducao-ao-docker-parte-3.md

```diff
@@ -0,0 +1,188 @@
+
+# Aplicação
+
+Não é o foco aqui mas vou mostrar rapidamente como funciona a aplicação. O código fonte se encontra [aqui](https://github.com/jvrmaia/flask-docker-example).
+
+```python
+import socket
+from flask import Flask
+from decouple import config
+
+app = Flask(__name__)
+
+VERSION = config('VERSION', 'dev', cast=str)
+
+@app.route("/")
+def index():
+    return socket.gethostname() + " - " + VERSION
+
+if __name__ == '__main__':
+    app.run(debug=True,host='0.0.0.0')
+```
+
+Basicamente, na raíz da aplicação retornamos o hostname e versão da aplicação e se a versão não for informada vai aparecer `dev`. Caso queira rodar local as instruções estão no README do repositório da aplicação.
+
+# Dockerfile
+
+O dockerfile é um documento de texto declarativo de construção da sua imagem Docker que será executado em ordem. Vamos ver o dockerfile do nosso projeto em questão:
+
+```dockerfile
+FROM python:3.6.3
+MAINTAINER Joao Maia "joao@joaovrmaia.com"
+
+COPY . /app
+
+WORKDIR /app
+
+RUN pip install -r requirements.txt
+
+ENV VERSION=0.0.1
+
+EXPOSE 5000
+
+ENTRYPOINT ["python"]
+
+CMD ["app.py"]
+```
+
+Vamos analisar cada item do Dockerfile:
+
+### FROM
+
+Essa instrução indica apartir de qual imagem a sua vai se basear, no nosso caso como vasmos subir uma aplicação Python já peguei uma imagem com o Python instalado para evitar esse processo. Estou usando uma das imagens oficiais do Docker. Nota, sempre que possível use imagens oficiais do Docker, evite usar imagem qualquer por ai na internet e verifique sempre as imagens que você for utilizar. Nada impede que uma pessoa crie uma imagem com software malicioso e divulgue publicamente.
+
+### MAINTAINER
+
+É muito comum encontrar essa entrada no Dockerfile mas ela não vai existir mais em breve. O recomendado agora é usar LABEL. Veja:
+
+```
+LABEL maintainer="joao@joaovrmaia.com"
+```
+
+### COPY
+
+O `COPY` copia arquivos do seu sistema de arquivos para o Docker bem parecido com o `cp` do terminal do linux. A diferença que ele aceita uma lista igual ao `CMD` do nosso exemplo, para alguns casos. Seu dever de casa é estudar o `ADD` e aprender as diferenças dele para o `COPY`. Observer que no nosso projeto de exemplo tem o arquivo `.dockerignore`, ele é parecido com o `.gitignore` só que para o Docker.
+
+### RUN
+
+Aqui você consegue rodar um ou mais comandos no seu Docker no tempo de criação, por exemplo, no nosso caso uso para instalar as dependências do nosso projeto.
+
+### ENV
+
+O `ENV` permite que você crie variáveis de ambientes padrão, no nosso caso estou colocando a versão da nossa imagem.
+
+### EXPOSE
+
+O `EXPOSE` permite que você torne visível portas do seu Docker e ainda escolha o protocolo, por padrão é o TCP. No caso de UDP seria:
```

Comment:
I think this line is wrongly saying that the application port will be visible in the host machine.
Don't really know how to improve it, but better rephrase it anyway